tap_postgres/sync_strategies/incremental.py

```diff
@@ -119,9 +119,48 @@ def sync_table(conn_info, stream, state, desired_columns, md_map):
                         singer.write_message(singer.StateMessage(value=copy.deepcopy(state)))
 
                     counter.increment()
+                
+            with conn.cursor(cursor_factory=psycopg2.extras.DictCursor, name='pipelinewise') as cur:
+                if rows_saved == 0:
+                    latest_sql = _get_select_latest_sql({"escaped_columns": escaped_columns,
+                                              "replication_key": replication_key,
+                                              "schema_name": schema_name,
+                                              "table_name": stream['table_name'],
+                                            })
+                    LOGGER.info('select latest row statement: %s', latest_sql)
+                    cur.execute(latest_sql)
+                    rec = cur.fetchone()
+                    if rec:
```

Comment:
(question) Just want to double-check, this logic will work well when the table is empty i.e. no row in the table, correct?

Reply:
According to `pyscopg`'s [doc](https://www.psycopg.org/docs/cursor.html), `fetchone` will return next row of a query result set, and will return `None` if no more data is available. 
In case the table is empty, when performing `cursor.fetchone()`, it will return `None`.  That's why I have the condition `if rec` at the line 133.